addons/mass_mailing/models/mass_mailing.py

```diff
@@ -408,230 +413,68 @@ def _get_statistics(self, cr, uid, ids, name, arg, context=None):
-        'failed': fields.function(
-            _get_statistics, string='Failed',
-            type='integer', multi='_get_statistics',
-        ),
-        'sent': fields.function(
-            _get_statistics, string='Sent',
-            type='integer', multi='_get_statistics',
-        ),
-        'delivered': fields.function(
-            _get_statistics, string='Delivered',
-            type='integer', multi='_get_statistics',
-        ),
-        'opened': fields.function(
-            _get_statistics, string='Opened',
-            type='integer', multi='_get_statistics',
-        ),
-        'replied': fields.function(
-            _get_statistics, string='Replied',
-            type='integer', multi='_get_statistics',
-        ),
-        'bounced': fields.function(
-            _get_statistics, string='Bounced',
-            type='integer', multi='_get_statistics',
-        ),
-        'failed': fields.function(
-            _get_statistics, string='Failed',
-            type='integer', multi='_get_statistics',
-        ),
-        'received_ratio': fields.function(
-            _get_statistics, string='Received Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'opened_ratio': fields.function(
-            _get_statistics, string='Opened Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'replied_ratio': fields.function(
-            _get_statistics, string='Replied Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'bounced_ratio': fields.function(
-            _get_statistics, String='Bouncded Ratio',
-            type='integer', multi='_get_statistics',
-        ),
-        'next_departure': fields.function(
-            _get_next_departure, string='Next Departure',
-            type='datetime'
-        ),
-    }
-
-    def mass_mailing_statistics_action(self, cr, uid, ids, context=None):
-        res = self.pool['ir.actions.act_window'].for_xml_id(cr, uid, 'mass_mailing', 'action_view_mass_mailing_statistics', context=context)
-        link_click_ids = self.pool['link.tracker.click'].search(cr, uid, [('mass_mailing_id', 'in', ids)], context=context)
+    @api.multi
+    def mass_mailing_statistics_action(self):
+        res = self.env.ref('mass_mailing.action_view_mass_mailing_statistics')
+        link_clicks_ids = self.env['link.tracker.click'].search([('mass_mailing_id', 'in', self.ids)]).ids
         res['domain'] = [('id', 'in', link_click_ids)]
         return res
 
-    def default_get(self, cr, uid, fields, context=None):
-        res = super(MassMailing, self).default_get(cr, uid, fields, context=context)
-        if 'reply_to_mode' in fields and not 'reply_to_mode' in res and res.get('mailing_model'):
-            if res['mailing_model'] in ['res.partner', 'mail.mass_mailing.contact']:
-                res['reply_to_mode'] = 'email'
+    @api.onchange('mass_mailing_campaign_id')
+    def onchange_mass_mailing_campaign_id(self):
```

Comment:
onchanges can be private methods. It becomes `onchange_mass_mailing_campaign_id`